fili-core/src/main/java/com/yahoo/bard/webservice/web/apirequest/DefaultHavingApiGenerator.java

```diff
@@ -24,16 +25,14 @@
 import java.util.Map;
 import java.util.Set;
 
-import static com.yahoo.bard.webservice.web.ErrorMessageFormat.HAVING_METRICS_NOT_IN_QUERY_FORMAT;
-
 /**
  * Generates having objects based on the having query in the api request.
  */
 public class DefaultHavingApiGenerator implements HavingGenerator {
     private static final Logger LOG = LoggerFactory.getLogger(DefaultHavingApiGenerator.class);
     private static final String COMMA_AFTER_BRACKET_PATTERN = "(?<=]),";
 
-    private final MetricDictionary metricDictionary;
+    private final Map<String, LogicalMetric> metricDictionary;
```

Comment:
Why are we changing type here?

Reply:
A very rather pedantic reason that came out of me being a bit anal retentive. Throughout the codebase, the `MetricDictionary` has a very specific (though unenforced) meaning: The collection of *all* configured metrics at any given scope. However, in order to fix the bug without complicating the `HavingGenerator` interface, we needed to pass in a subset of the configured metrics: those that were requested by the query. I didn't like referring to that as a `MetricDictionary`, because I felt it violated the principal of least surprise. So we loosened the requirements down to a map of metrics instead.